instrumentation/opentelemetry-instrumentation-asyncio/tests/test_asyncio_future_cancellation.py

```diff
@@ -0,0 +1,63 @@
+import asyncio
+from unittest.mock import patch
+
+import pytest
+
+from opentelemetry.instrumentation.asyncio import AsyncioInstrumentor
+from opentelemetry.instrumentation.asyncio.environment_variables import (
+    OTEL_PYTHON_ASYNCIO_FUTURE_TRACE_ENABLED,
+)
+from opentelemetry.test.test_base import TestBase
+from opentelemetry.trace import get_tracer
+
+
+class TestTraceFuture(TestBase):
+    @patch.dict(
+        "os.environ", {OTEL_PYTHON_ASYNCIO_FUTURE_TRACE_ENABLED: "true"}
+    )
+    def setUp(self):
+        super().setUp()
+        self._tracer = get_tracer(
+            __name__,
+        )
+        self.instrumentor = AsyncioInstrumentor()
+        self.instrumentor.instrument()
+
+    def tearDown(self):
+        super().tearDown()
+        self.instrumentor.uninstrument()
+
+    @pytest.mark.asyncio
+    def test_trace_future_cancelled(self):
+        async def future_cancelled():
+            with self._tracer.start_as_current_span("root"):
+                future = asyncio.Future()
+                future = self.instrumentor.trace_future(future)
+                future.cancel()
+
+        try:
+            asyncio.run(future_cancelled())
+        except asyncio.CancelledError as exc:
+            self.assertEqual(isinstance(exc, asyncio.CancelledError), True)
```

Comment:
With `pytest.mark.asyncio` I was thinking you could do this 
```suggestion
    async def test_trace_future_cancelled(self):
        with self._tracer.start_as_current_span("root"):
            future = asyncio.Future()
            future = self.instrumentor.trace_future(future)
            future.cancel()

        try:
            await future
        except asyncio.CancelledError as exc:
            self.assertEqual(isinstance(exc, asyncio.CancelledError), True)
```

Does that work?

Reply:
That code seems to ignore the assert.
If you modify it with this code, and replace the 2 below with a 1, it will pass.
```
        self.assertEqual(len(spans), 2)
        self.assertEqual(len(metrics), 2)
```

Does the test work?

Reply:
It sounds like pytest-asyncio is not installed then. You can just leave teh code as is but please remove the `@pytest.mark.asyncio` as it's a bit misleading

Reply:
i removed it. Thank you for your feedback.